skopt/tree_opt.py

```diff
@@ -0,0 +1,255 @@
+    res.func_vals = yi
+    res.x_iters = Xi
+    res.models = models
+
+    return res
+
+
+def gbrt_minimize(func, dimensions, base_estimator=None, maxiter=100,
+                  n_points=20, n_start=10, random_state=None):
+    """Sequential optimisation using gradient boosted trees.
+
+    Gradient boosted regression trees are used to model the (very)
+    expensive to evaluate function `func`. The model is improved
+    by sequentially evaluating the expensive function at the next
+    best point. Thereby finding the minimum of `func` with as
+    few evaluations as possible.
+
+    Parameters
+    ----------
+    * `func` [callable]:
+        Function to minimize. Should take a array of parameters and
+        return the function values.
+
+    * `dimensions` [list, shape=(n_dims,)]:
+        List of search space dimensions.
+        Each search dimension can be defined either as
+
+        - a `(upper_bound, lower_bound)` tuple (for `Real` or `Integer`
+          dimensions),
+        - a `(upper_bound, lower_bound, "prior")` tuple (for `Real`
+          dimensions),
+        - as a list of categories (for `Categorical` dimensions), or
+        - an instance of a `Dimension` object (`Real`, `Integer` or
+          `Categorical`).
+
+    * `base_estimator` [`GradientBoostingQuantileRegressor`]:
+        The regressor to use as surrogate model
+
+    * `maxiter` [int, default=100]:
+        Number of iterations used to find the minimum. This corresponds
+        to the total number of evaluations of `func`. If `n_start` > 0
+        only `maxiter - n_start` iterations are used.
+
+    * `n_start` [int, default=10]:
+        Number of random points to draw before fitting `base_estimator`
+        for the first time. If `n_start > maxiter` this degrades to
+        a random search for the minimum.
+
+    * `n_points` [int, default=20]:
+        Number of points to sample when minimizing the acquisition function.
+
+    * `random_state` [int, RandomState instance, or None (default)]:
+        Set random state to something other than None for reproducible
+        results.
+
+    Returns
+    -------
+    * `res` [`OptimizeResult`, scipy object]:
+        The optimization result returned as a OptimizeResult object.
+        Important attributes are:
+
+        - `x` [float]: location of the minimum.
+        - `fun` [float]: function value at the minimum.
+        - `models`: surrogate models used for each iteration.
+        - `x_iters` [array]: location of function evaluation for each
+           iteration.
+        - `func_vals` [array]: function value for each iteration.
+
+        For more details related to the OptimizeResult object, refer
+        http://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.OptimizeResult.html
+    """
+    rng = check_random_state(random_state)
+
+    # Default estimator
+    if base_estimator is None:
+        gbrt = GradientBoostingRegressor(n_estimators=20, loss='quantile')
```

Comment:
Should this be set here or moved to `GradientBoostingQuantileRegressor`? I can't make up my mind, please help :)